lib/src/benchmark_base.dart

```diff
@@ -35,16 +35,21 @@ class BenchmarkBase {
   // time minimum has been reached.
   static double measureFor(Function f, int minimumMillis) {
     int minimumMicros = minimumMillis * 1000;
-    int iter = 0;
+    int iter = 1;
+    int elapsed = 0;
     Stopwatch watch = Stopwatch();
     watch.start();
-    int elapsed = 0;
-    while (elapsed < minimumMicros) {
-      f();
+    while (true) { // ignore: literal_only_boolean_expressions
+      watch.reset();
+      for (int i = 0; i < iter; i++) {
+        f();
+      }
       elapsed = watch.elapsedMicroseconds;
-      iter++;
+      if (elapsed >= minimumMicros) {
+        return elapsed / iter;
+      }
+      iter *= 2;
```

Comment:
Is the logic correct here?
In the first iteration of the while-loop, iter is one, you run once. If it stops, there, iter is correct.
Then you multiply iter by 2, and restart the loop. After the for loop, you have run `f` a total of 3 times, but iter is only 2.

I think you'll need separate loop-size and total-iter variables (or a cleverer computation at the end, like `var total = iter * 2 - 1; return elapsed / total;`).

Reply:
I think the logic is correct, but the code could be a little clearer. The important bit is that the stopwatch gets reset at the top of the outer loop, so the elapsed time from the previous run is discarded.

I think a cleaner version is:

```dart
  static double measureFor(Function f, int minimumMillis) {
    int minimumMicros = minimumMillis * 1000;
    int iter = 1;
    while (true) { // ignore: literal_only_boolean_expressions
      var watch = Stopwatch()..start();
      for (int i = 0; i < iter; i++) {
        f();
      }
      var elapsed = watch.elapsedMicroseconds;
      if (elapsed >= minimumMicros) {
        return elapsed / iter;
      }
      iter *= 2;
    }
```

Reply:
This still discards half the runs. That's a lot of overhead. I'd do:
```dart
static double measureFor(Function f, int minimumMillis) {
  int minimumMicros = minimumMillis * 1000;
  int iter = 1;
  int elapsed = 0;
  var watch = Stopwatch()..start();
  do {
    for (int i = 0; i < iter; i++) {
      f();
    }
    elapsed = watch.elapsedMicroseconds;
    iter *= 2;
  } while (elapsed < minimumMicros);
  return elapsed / (iter * 2 - 1);
}
```
This counts all the calls to `f()`. The overhead of doing `iter *= 2;` and `elapsed < minimumMicrods` should not be measurable in comparison (only happens a logarithmic number of times), but the total of calls to `f()` are.

(It will still run for an average of 1.5 times the minimum time)